MergerLogicUnitTests/ImageProcessing/TileScalerTest.cs

```diff
@@ -90,17 +89,39 @@ public static IEnumerable<object[]> GetUpscaleTilesTestParameters()
                 new Coord(18, 0, 0),
                 File.ReadAllBytes("5_upscaled_16_18.png"),
             };
+
+            yield return new object[] {
+                File.ReadAllBytes("3.jpeg"),
+                new Coord(9, 0, 0),
+                new Coord(10, 0, 0),
+                File.ReadAllBytes("3_upscaled_9_10.jpeg"),
+            };
+
+
+            yield return new object?[] {
+                File.ReadAllBytes("empty_tile.png"),
+                new Coord(3, 0, 0),
+                new Coord(4, 0, 0),
+                null,
+            };
         }
         [TestMethod]
         [TestCategory("Upscale")]
         [DynamicData(nameof(GetUpscaleTilesTestParameters), DynamicDataSourceType.Method)]
-        public void Upscale(byte[] tileBytes, Coord baseTileCoord, Coord targetCoord, byte[] expectedTileBytes)
+        public void Upscale(byte[] tileBytes, Coord baseTileCoord, Coord targetCoord, byte[]? expectedTileBytes)
         {
             var testTile = new Tile(baseTileCoord, tileBytes);
             var resultTile = this._testTileScaler.Upscale(testTile, targetCoord);
 
-            Assert.IsNotNull(resultTile);
-            CollectionAssert.AreEqual(expectedTileBytes, resultTile.GetImageBytes());
+            if (expectedTileBytes is null)
```

Comment:
I'm not sure this is the best approach, maybe we should add a parameter indicating we expect the result to be `null`. This can happen when we get a `PNG` image too.